packages/components/components/Button/Button.tsx

```diff
@@ -1,207 +1,161 @@
-import React, { FC } from "react";
+import React from "react";
 import { styled } from "../../stitches.config";
 
 export interface IButtonProps {
   as?: "button" | "a";
-  type?: "primary" | "secondary" | "tertiary" | "accent1" | "accent2";
-  outline?: boolean;
-  ghost?: boolean;
+  type?: "primary" | "secondary" | "link";
+  tone?: "forest" | "wheat";
   leftIcon?: React.ReactNode;
   rightIcon?: React.ReactNode;
   children?: string | React.ReactNode;
   href?: string;
+  tabIndex?: number;
+  disabled?: boolean;
 }
 
-const Button = ({ leftIcon, rightIcon, children, ...props }: IButtonProps) => {
-  return <ButtonContainer {...props}>
-    {leftIcon}
+const Button = ({ leftIcon, rightIcon, children, as = "button", ...props }: IButtonProps) => {
+  return <ButtonContainer as={as} {...props}>
+    {leftIcon && <Icon left>{leftIcon}</Icon>}
     {children}
-    {rightIcon}
+    {rightIcon && <Icon right>{rightIcon}</Icon>}
   </ButtonContainer>
 }
 
 const ButtonContainer = styled("button", {
-  display: "flex",
+  display: "inline-flex",
+  flexGrow: 0,
   justifyContent: "center",
   alignItems: "center",
   fontFamily: "$sans",
   fontWeight: 600,
   border: "none",
   fontSize: "$sm",
-  padding: "$4 $10",
   transition: "all 0.2s ease-in-out",
   textDecoration: "none",
-  svg: { 
-    height: '$5',
-    width: '$5',
-    mr: '$2',
-    position: 'relative',
-    left: '0',
-  },
-  "svg:first-child": {
-    mr: "$2",
-  },
-  "svg:last-child": {
-    ml: "$2",
-  },
+  boxSizing: "border-box",
+  cursor: "pointer",
   defaultVariants: {
     type: "primary",
+    tone: "forest",
   },
   variants: {
     type: {
       primary: {
-        backgroundColor: "$primary",
-        color: "$sand",
+        border: "none",
+        height: '$10',
+        py: 0,
+        px: '$4',
       },
       secondary: {
-        backgroundColor: "$secondary",
-        color: "$forest",
-      },
-      tertiary: {
-        backgroundColor: "$tertiary",
-        color: "$forest",
-      },
-      accent1: {
-        backgroundColor: "$accent1",
-        color: "$forest",
+        borderStyle: "solid",
+        borderWidth: "1px",
+        height: '$10',
+        py: 0,
+        px: '$4',
       },
-      accent2: {
-        backgroundColor: "$accent2",
-        color: "$forest",
+      link: {
+        borderStyle: "solid",
+        borderWidth: "1px",
+        background: "transparent",
+        border: "none",
+        p: 0,
       },
     },
-    outline: {
-      true: {
-        boxSizing: "border-box",
+    tone: {
+      forest: {
+      },
+      wheat: {
```

Comment:
should these not have some colors or something with it? why is it empty?

Reply:
I was getting typescript errors without having them defined as variants even though all their styles are later defined under compoundVariants. Not really sure if theres a better way to handle it.